src/components/policy/policy_external/include/policy/status.h

```diff
@@ -0,0 +1,200 @@
+/*
+ Copyright (c) 2016, Ford Motor Company
+ All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are met:
+
+ Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+
+ Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following
+ disclaimer in the documentation and/or other materials provided with the
+ distribution.
+
+ Neither the name of the Ford Motor Company nor the names of its contributors
+ may be used to endorse or promote products derived from this software
+ without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SRC_COMPONENTS_POLICY_POLICY_EXTERNAL_INCLUDE_POLICY_STATUS_H_
+#define SRC_COMPONENTS_POLICY_POLICY_EXTERNAL_INCLUDE_POLICY_STATUS_H_
+
+#include <string>
+#include "policy/policy_types.h"
+#include "utils/macro.h"
+
+namespace policy {
+
+/**
+ * @brief The UpdateEvent enum defines system events which can change update
+ * status
+ */
+enum UpdateEvent {
+  kOnNewAppRegistered,
+  kOnValidUpdateReceived,
+  kOnWrongUpdateReceived,
+  kOnUpdateSentOut,
+  kOnUpdateTimeout,
+  kOnResetPolicyTableRequireUpdate,
+  kOnResetPolicyTableNoUpdate,
+  kScheduleUpdate,
+  kOnResetRetrySequence
+};
+
+class UpdateStatusManager;
+
+/**
+ * @brief The Status class defines base implementation of update status
+ */
+class Status {
```

Comment:
@dev-gh isn't this better to implement this class as following

1. move the status string and status enum as a member of base class
2. Initialize string and enum value in the descendant class constructor
3. Don't override getStatus and getStatusString() at all.

So it might look as 

```c++
class Status {
public:
std::string get_status_string() {return stringified_status_; }
PolicyTableStatus get_status() {return status_;}

protected:
 Status(PolicyTableStatus status, std::string stringified_status) // or create map
   : stringified_status_(stringified_status),
     status_(status){}
private:
std::string stringified_status_;
PolicyTableStatus  status_;
}

class UpToDate: public Status {
UpToDate(): Status(StatusUpToDate, "UP_TO_DATE"){}

}
```

So basically only `ProcessEvent` will be overriden